tools/salmon-kallisto-mtx-to-10x/salmonKallistoMtxTo10x.xml

```diff
@@ -1,15 +1,22 @@
-<tool id="_salmon_kallisto_mtx_to_10x" name="salmonKallistoMtxTo10x" version="0.0.1+galaxy0">
+<tool id="_salmon_kallisto_mtx_to_10x" name="salmonKallistoMtxTo10x" version="0.0.1+galaxy1">
     <description>Transforms .mtx matrix and associated labels into a format compatible with tools expecting old-style 10X data</description>
     <requirements>
       <requirement type="package">scipy</requirement>
       <requirement type="package">pandas</requirement>
     </requirements>
-    <command interpreter="python" detect_errors="exit_code"><![CDATA[
-        $__tool_directory__/salmonKallistoMtxTo10x.py --cell_prefix "${cell_prefix}" "${mtx_file}" "${genes_file}" "${barcodes_file}" ./
+    <command detect_errors="exit_code"><![CDATA[
+        file $mtx_file | grep 'gzip compressed' > /dev/null;
+        if [ $? -eq 0 ]; then
+            matrixfile=matrix.mtx.gz;
+        else
+            matrixfile=matrix.mtx;
+        fi;
+        ln -s ${mtx_file} \$matrixfile;
+        $__tool_directory__/salmonKallistoMtxTo10x.py --cell_prefix "${cell_prefix}" "\${matrixfile}" "${genes_file}" "${barcodes_file}" ./
 	    ]]></command>
 
     <inputs>
-        <param name="mtx_file" type="data" format="txt" label=".mtx-format matrix" />
+        <param name="mtx_file" type="data" format="mtx" label=".mtx-format matrix" />
```

Comment:
I wonder if one would add here `format="mtx,mtx.gz"`. What do you think @nh3 ?

Reply:
It depends on whether galaxy supports that as a valid data type. As I just checked, the version running at usegalaxy.org doesn't. If the newer version supports it, then it can be handy.